### ISSUE TITLE
Check remote changes periodically

### ISSUE THREAD

**Issue by Riajyuu** [gitextensions/gitextensions#7099]
## Feature description

Check if there is any change on current branch at remote/origin periodically and warn users if any.

Perhaps check other branches as well?

## Environment

Irrelevant

**Comment by mast-eu** [comment]
We already have a plugin for this: `Plugins` > `Periodic background fetch`.

See https://git-extensions-documentation.readthedocs.io/en/latest/plugins.html#periodic-background-fetch

With this you should be able to achieve what you want.

**Comment by mast-eu** [comment]
Closing, because already implemented.

**Comment by Riajyuu** [comment]
> See https://git-extensions-documentation.readthedocs.io/en/latest/plugins.html#periodic-background-fetch
> 
> With this you should be able to achieve what you want.

It is not. Sometimes I may not accept remote changes: I simply wanna know what happened

**Comment by mast-eu** [comment]
Please explain in more detail how you see the check to be implemented, if not with git fetch?
How do you expect to be notified about remote changes?
How do you expect to see them, in order to decide whether to accept them or not?
How should accepting/refusing work?

**Comment by Riajyuu** [comment]
> how you see the check to be implemented, if not with git fetch?

git commit log so that you know how many commits left remote

>How do you expect to be notified about remote changes?

1. any remote changes? And if any,
2. how many commits.

>How do you expect to see them, in order to decide whether to accept them or not?

On toolbar button use a indicator just like GitHub Desktop?

>How should accepting/refusing work?

Accept simply means fetch & pull. Can't refuse since it is only about remote branch info ( it has happened)

**Comment by pmiossec** [comment]
@Riajyuu Your feature request is not clear.

I agree with @mast-eu , the only way to know if there are changes in a remote branch is to fetch it. So, the "periodic background fetch" is the answer to your feature.

>how many commits.

and 

>On toolbar button use a indicator just like GitHub Desktop?

Could you provide a screenshot ? Because I don't understand what is different from what GitExtensions already display with commits ahead/behind (done in #5994)

> GitHub Desktop checks every 10 min without any option

It's, in my opinion, a bad practice to automatically fetch by default because that "breaks" the safe-guard introduced by the command `git push --force-with-lease`. So, for me, that should be an option enabled by the user. Like that, he is well aware that it could loose commit even when using `--force-with-lease`

To sum up, I agree with @mast-eu . Everything is here to do what you want. If we miss something, feel free to open a more specific feature request (or update the title of this one to provide a more accurate description)

**Comment by gerhardol** [comment]
If GH desktop is not using git-fetch, it could use a private extension that GE cannot use anyway.

**Comment by mstv** [comment]
> Sometimes I may not accept remote changes: I simply wanna know what happened.

That is what `fetch` is intended for. `Fetch` does not change anything in your working copy and commit index. If you don't want to accept the changes, feel free to not use them or to reset the branch to the last commit you agree with.
